runtime/lua/vim/lsp/util.lua

```diff
@@ -2223,6 +2112,18 @@ function M.make_formatting_params(options)
   }
 end
 
+--- Returns the buffer number for the given {bufnr}.
+---
+---@param bufnr (integer|nil) Buffer number to resolve. Defaults to current buffer
+---@return integer bufnr
+function M.resolve_bufnr(bufnr)
+  vim.validate({ bufnr = { bufnr, 'n', true } })
+  if bufnr == nil or bufnr == 0 then
+    return api.nvim_get_current_buf()
+  end
+  return bufnr
+end
+
```

Comment:
We want to get rid of `vim.lsp.util` completely, not dump more stuff in there...

Reply:
okay this is not relate actually ..

Reply:
Pro tip: the less changes you make in a PR, the faster it is to review it.

Reply:
This includes _not_ adding any autocompletion (yet). Please keep _this_ PR to a pure refactor, and then make a separate `feat` PR for the autocompletion afterwards (which I'm not 100% certain we want, personally, but it's @mfussenegger's call).

Reply:
Is it really too much to ask for to make your enthusiastic changes in smaller, targeted, correctly labelled, PRs?